core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -275,6 +285,13 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
     responseListeners ::= onResponse
   }
 
+  def updateErrorMetrics(apiKey: ApiKeys, errors: scala.collection.Map[Errors, Integer]) {
+    errors.foreach { case (error, count) =>
+      val errorMeter = RequestMetrics.metricsMap(apiKey.name).errorMeters(error)
```

Comment:
Maybe there should be a method in the `RequestMetrics` class called `markErrorMeter` which takes care of the `getOrCreate`.